swApiClient.cpp

```diff
@@ -13,34 +30,80 @@ swApiClient Constructor
   Constructs swApiClient
 ====================
 */
-swApiClient::swApiClient( std::string cKey, std::string cSecret, std::string cbUrl) {
+swApiClient::swApiClient( string cKey, string cSecret) {
     consumerKey = cKey;
     consumerSecret = cSecret;
-    callbackUrl = cbUrl;
+    accessToken = "";
 }
 
 /*
 ====================
-swApiClient Url
+swApiClient Authorize
 
-  Generate the full URL for an API path 
+  Connect to the Shapeways API using oauth2
 ====================
 */
-std::string swApiClient::Url( std::string path ) {
-    if ( path[0] != '/' ) {
-        path = '/' + path;
-    }
-    if ( path[path.length() - 1] != '/' ) {
-        path += '/';
-    }
+void swApiClient::Authorize() {
 
-    return BASE_URL + path + API_VERSION;
+  CURL *curl;
+  CURLcode res;
+  string readBuffer;
+
+  // prepare auth payload
+  char *postData = (char*)malloc(250);
+  strcpy(postData, "grant_type=client_credentials&client_id=");
+  strcat(postData, this->consumerKey.c_str());
+  strcat(postData, "&client_secret=");
+  strcat(postData, this->consumerSecret.c_str());
+
+  curl = curl_easy_init();
```

Comment:
cool kids just do `CURL *curl = curl_easy_init();` :) but those same cool kids also participate in the worst readable C code I think

Reply:
aye, let's keep this two lines as is. More clear.